tfe/resource_tfe_organization_run_task.go

```diff
@@ -0,0 +1,185 @@
+func resourceTFEOrganizationRunTaskCreate(d *schema.ResourceData, meta interface{}) error {
+	tfeClient := meta.(*tfe.Client)
+
+	// Get the task name and organization.
+	name := d.Get("name").(string)
+	organization := d.Get("organization").(string)
+
+	// Create a new options struct.
+	options := tfe.RunTaskCreateOptions{
+		Name:     name,
+		URL:      d.Get("url").(string),
+		Category: d.Get("category").(string),
+		HMACKey:  tfe.String(d.Get("hmac_key").(string)),
+		Enabled:  tfe.Bool(d.Get("enabled").(bool)),
+	}
+
+	log.Printf("[DEBUG] Create task %s for organization: %s", name, organization)
+	task, err := tfeClient.RunTasks.Create(ctx, organization, options)
+	if err != nil {
+		return fmt.Errorf(
+			"Error creating task %s for organization %s: %w", name, organization, err)
+	}
+
+	d.SetId(task.ID)
+
+	return resourceTFEOrganizationRunTaskRead(d, meta)
+}
+
+func resourceTFEOrganizationRunTaskDelete(d *schema.ResourceData, meta interface{}) error {
+	tfeClient := meta.(*tfe.Client)
+
+	log.Printf("[DEBUG] Delete task: %s", d.Id())
+	err := tfeClient.RunTasks.Delete(ctx, d.Id())
+	if err != nil {
+		if isErrResourceNotFound(err) {
+			return nil
+		}
+		return fmt.Errorf("Error deleting task %s: %w", d.Id(), err)
+	}
+
+	return nil
+}
+
+func resourceTFEOrganizationRunTaskUpdate(d *schema.ResourceData, meta interface{}) error {
+	tfeClient := meta.(*tfe.Client)
+
+	// Setup the options struct
+	options := tfe.RunTaskUpdateOptions{}
+	if d.HasChange("name") {
+		options.Name = tfe.String(d.Get("name").(string))
+	}
+	if d.HasChange("url") {
+		options.URL = tfe.String(d.Get("url").(string))
+	}
+	if d.HasChange("category") {
+		options.Category = tfe.String(d.Get("category").(string))
+	}
+	if d.HasChange("enabled") {
+		options.Enabled = tfe.Bool(d.Get("enabled").(bool))
+	}
+	if d.HasChange("hmac_key") {
+		options.HMACKey = tfe.String(d.Get("hmac_key").(string))
+	}
+
+	log.Printf("[DEBUG] Update configuration of task: %s", d.Id())
+	task, err := tfeClient.RunTasks.Update(ctx, d.Id(), options)
+	if err != nil {
+		return fmt.Errorf("Error updating task %s: %w", d.Id(), err)
+	}
+
+	d.SetId(task.ID)
+
+	return resourceTFEOrganizationRunTaskRead(d, meta)
+}
+
+func resourceTFEOrganizationRunTaskRead(d *schema.ResourceData, meta interface{}) error {
+	tfeClient := meta.(*tfe.Client)
+
+	log.Printf("[DEBUG] Read configuration of task: %s", d.Id())
+	task, err := tfeClient.RunTasks.Read(ctx, d.Id())
+
+	if err != nil {
+		if isErrResourceNotFound(err) {
+			log.Printf("[DEBUG] Task %s does not exist", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading configuration of task %s: %w", d.Id(), err)
+	}
+
+	// Update the config.
+	d.Set("name", task.Name)
+	d.Set("url", task.URL)
+	d.Set("category", task.Category)
+	d.Set("enabled", task.Enabled)
+	// The HMAC Key is always empty from the API so all we can do is
+	// echo the request's key to the response
+	d.Set("hmac_key", tfe.String(d.Get("hmac_key").(string)))
```

Comment:
null [can also be returned](https://www.terraform.io/cloud-docs/api-docs/run-tasks#sample-response)... the tests may be hiding this since the provider default is "" ?

Reply:
This is the same conversation as above.  Resolving for now.